### PR TITLE
Fix issue with interceptor instances being created multiple times for same bean

### LINKED ISSUE #3530: Intercepted subclass does not share interceptor instances
"The lifecycle of an interceptor instance is the same as that of the target class instance with which it is associated."

### PR DESCRIPTION
**Pull request by manovotn**
Included some minor code cleanup - unused method and outdated TODO.

Fixes #3530

### CODE REVIEW

**Comment by manovotn** [comment]
So, since Martin is on PTO, who's brave enough to review a PR with some Gizmo code generation? :)

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java [hunk 1]
```diff
@@ -138,9 +138,35 @@ protected FieldDescriptor createConstructor(ClassOutput classOutput, BeanInfo be
...
         Map<InterceptorInfo, ResultHandle> interceptorToResultHandle = new HashMap<>();
         for (int j = 0; j < boundInterceptors.size(); j++) {
             interceptorToResultHandle.put(boundInterceptors.get(j), constructor.getMethodParam(j + superParamsSize + 1));
+
+            // create instance of each interceptor -> InjectableInterceptor.get()
+            ResultHandle creationalContext = constructor.invokeStaticMethod(MethodDescriptors.CREATIONAL_CTX_CHILD,
+                    creationalContextHandle);
+            ResultHandle interceptorInstance = constructor.invokeInterfaceMethod(
+                    MethodDescriptors.INJECTABLE_REF_PROVIDER_GET,
+                    constructor.getMethodParam(j + superParamsSize + 1), creationalContext);
```
Comment (gsmet) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: Maybe it's worth creating a `ResultHandle` for this one? It's used 3 times AFAICS.

**Review by gsmet — CHANGES_REQUESTED** [review]
The code looks good apart from my minor comment. Can you see if you can take care of it?

Not a blocker though.

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java (hunk 1 above)
Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: Good point, fixed!

**Review by gsmet — APPROVED** [review]

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java [hunk 2]
```diff
@@ -138,9 +138,35 @@ protected FieldDescriptor createConstructor(ClassOutput classOutput, BeanInfo be
                         parameterTypes.subList(0, superParamsSize).toArray(new String[0])),
                 constructor.getThis(), superParams);
 
+        // we build a map for each interceptor instance created, so that they are shared
+        // Map<String, Object> where InjectableInterceptor.getIdentifier() is key and Object is instance of the interceptor for this bean
+        FieldCreator interceptorInstanceMap = subclass.getFieldCreator("interceptorInstanceMap", Map.class.getName())
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: I think there is no need to add the map as a class field. It should be a local variable instead...

**Pull request merged**